src/main/java/com/scottlogic/AuthorPostSorter.java

```diff
@@ -1,16 +1,22 @@
 package com.scottlogic;
 
+import java.util.ArrayList;
 import java.util.Collections;
 import java.util.Comparator;
 import java.util.List;
 
 public class AuthorPostSorter implements PostSorter {
 
     @Override
-    public List<UserPost> sort(List<UserPost> inputList) {
-        if(inputList != null){
-                inputList.sort(Comparator.comparing(UserPost::getAuthor));
+    public List<UserPost> sort(List<UserPost> inputList, SortOrder SortOrder) {
+        if (inputList != null) {
+            List<UserPost> copyInputList = new ArrayList<>(inputList);
```

Comment:
`copyInputList` is a good name to begin with but after a few lines the naming breaks down a little as it implies that this will always be a copy of the input list which isn't true. Maybe something like `mutableInputList`, `sortableList`, `outputList` or something that is true throughout the programme might be a better shout.